strutil/pathiter.go

```diff
@@ -66,60 +66,73 @@ func (iter *PathIterator) Path() string {
 	return iter.path
 }
 
-// CurrentBase returns the name of the current path element.
-// It will only end with a slash if the iterator is currently
-// placed at the base of the filesystem.
-// Ex:
+// CurrentBase returns the last element of the current path.
+// It removes any trailing slash unless the last element of the
+// current path is the filesystem root (and so CurrentBase() == "/").
```

Comment:
Nitpick:  `(and so CurrentBase() == "/")` -> I think this is too much and not required